receiver.go

```diff
@@ -351,13 +361,11 @@ func (r *Receiver) listenForMessage(ctx context.Context) (*amqp.Message, error)
 }
 
 // newSessionAndLink will replace the session and link on the Receiver
+// NOTE: this does *not* take the write lock, callers must hold it as required!
 func (r *Receiver) newSessionAndLink(ctx context.Context) error {
 	ctx, span := r.startConsumerSpanFromContext(ctx, "sb.Receiver.newSessionAndLink")
 	defer span.End()
 
-	r.clientMu.Lock()
```

Comment:
Just checking since this is being removed now, are there any other calls to newSessionAndLink elsewhere that need to take the lock now? Or is it just in Recover?

Reply:
It's just in recover.  The only other place where `newSessionAndLink()` is called is when creating the receiver so it can't race with any other calls.